srv/salt/ceph/dashboard/default.sls

```diff
@@ -0,0 +1,32 @@
+{% set dashboard_user = salt['pillar.get']('dashboard_user', 'admin') %}
+{% set dashboard_pw = salt['pillar.get']('dashboard_password', salt['grains.get']('dashboard_creds:' ~ dashboard_user , salt['random.get_str'](10))) %}
```

Comment:
Why the grains lookup here? Iiuc having a password in the grains means we've created the user already.

Reply:
There needs to be a password for the previously specified user though. The password being present indicates that this user is already registered.

Reply:
> There needs to be a password for the previously specified user though.

I don't follow why.

> The password being present indicates that this user is already registered.

Right and that means that both `set username and password` and `set dashboard password grain` won't be executed...?

Reply:
If you don't do the grain lookup here, a default password will be printed to the screen even though you already chose a password.

```
admin:~ # cat /srv/pillar/ceph/stack/global.yml
dashboard_user: test3
dashboard_password: test3


admin:~ # salt \admin* state.apply ceph.dashboard
admin:
  Name: ceph mgr module enable dashboard - Function: cmd.run - Result: Changed Started: - 09:38:38.273895 Duration: 376.823 ms                                                                
  Name: ceph dashboard create-self-signed-cert - Function: cmd.run - Result: Changed Started: - 09:38:38.650988 Duration: 282.95 ms                                                           
  Name: /bin/true - Function: cmd.run - Result: Clean Started: - 09:38:38.934131 Duration: 252.67 ms                                                                                          
  Name: ceph dashboard ac-user-create test3 test3 administrator - Function: cmd.run - Result: Clean Started: - 09:38:39.187332 Duration: 0.003 ms                                             
  Name: grains.set - Function: module.run - Result: Clean Started: - 09:38:39.188225 Duration: 0.004 ms  
```

if you now change the password in the pillar (or delete it), the grain lookup will still print to correct password to the screen. Removing the grain lookup will generate a new one although it will not be saved in the grains.


Writing this makes me wonder if upon password removal in the pillar, we should react to that.
Overwriting it is possbile with the available `ceph dashboard user foo ` calls..

Reply:
> Writing this makes me wonder if upon password removal in the pillar, we should react to that.
> Overwriting it is possbile with the available `ceph dashboard user foo ` calls..

Well.. That'd probably be even more confusing when a user changes the password in the web-ui and the password will be reset on the next stage.x invocation. This should probably be a one-shot operation